### PR TITLE
feat: updates to widget toolbar

### PR DESCRIPTION
**Pull request by just-toby**
design spec: https://www.figma.com/file/kNSDBMpOzxSTOP6MerohLm/Web-Design-Spec?node-id=10305%3A105759&t=gY6rvL8clpqdZ5EE-4

moves the Approval/Allowance UI from the ActionButton to the Toolbar. it isn't matching the design spec exactly yet, but this is where it should live. i'll update the styling in a follow up

also, removes some props from the different Button types. these components are all getting the same value which can come from a shared hook.

also note that we hide the toolbar if either the input or output currency is unset, per the design

### CODE REVIEW

File: src/components/Swap/SwapActionButton/SwapButton.tsx [hunk 1]
```diff
@@ -12,29 +12,19 @@ import { useUniversalRouterSwapCallback } from 'hooks/useUniversalRouter'
...
-import { Colors } from 'theme'
+import { TransactionType } from 'state/transactions'
 import invariant from 'tiny-invariant'
 
 import ActionButton from '../../ActionButton'
 import Dialog from '../../Dialog'
 import { SummaryDialog } from '../Summary'
-import AllowanceButton from './AllowanceButton'
-import ApproveButton from './ApproveButton'
+import useOnSubmit from './useOnSubmit'
```
[the remark this answers is not in the document]
Comment (just-toby) [reply src/components/Swap/SwapActionButton/SwapButton.tsx:21]: updated Approve Button

File: src/components/Swap/SwapActionButton/SwapButton.tsx [hunk 2]
```diff
@@ -106,16 +98,6 @@ export default function SwapButton({
     setOpen(await onReviewSwapClick())
   }, [onReviewSwapClick])
 
-  if (permit2Enabled) {
```
[the remark this answers is not in the document]
Comment (just-toby) [reply src/components/Swap/SwapActionButton/SwapButton.tsx:109]: pulled the change from #358 into this PR

File: src/components/Swap/Toolbar/index.tsx [hunk 3]
```diff
@@ -76,8 +87,22 @@ export default memo(function Toolbar() {
     impact,
   ])
 
+  if (inputCurrency == null || outputCurrency == null) {
+    return null
+  }
+
+  if (permit2Enabled) {
```
[the remark this answers is not in the document]
Comment (just-toby) [reply src/components/Swap/Toolbar/index.tsx:92]: no, i mentioned this in the PR description - i still need to tweak the styling here a bit

[the remark this answers is not in the document]
Comment (just-toby) [reply src/components/Swap/Toolbar/index.tsx:92]: they look "fine" though, just not matching the spec yet

File: src/components/Swap/SwapActionButton/AllowanceButton.tsx [hunk 4]
```diff
@@ -3,21 +3,17 @@ import { PERMIT2_ADDRESS } from '@uniswap/permit2-sdk'
 import ActionButton from 'components/ActionButton'
 import EtherscanLink from 'components/EtherscanLink'
 import { usePendingApproval } from 'hooks/transactions'
+import useButtonColor from 'hooks/useTokenColorExtraction'
```
Comment (zzmp) [inline src/components/Swap/SwapActionButton/AllowanceButton.tsx:6]: Run `yarn lint --fix` to get these sorted

File: src/components/Swap/SwapActionButton/AllowanceButton.tsx [hunk 5]
```diff
@@ -72,7 +68,7 @@ export default function AllowanceButton({ token, isApprovalLoading, approveAndPe
   }, [isApprovalLoading, isFailed, isPending, onClick, pendingApproval, token?.symbol])
 
   return (
-    <ActionButton color={color} disabled={!action?.onClick} action={action}>
+    <ActionButton color={useButtonColor()} disabled={!action?.onClick} action={action}>
```
Comment (zzmp) [inline src/components/Swap/SwapActionButton/AllowanceButton.tsx:71]: Is ActionButton ever used with a color that is _not_ `useButtonColor`? Or can it be updated to have the color be internal to the component?

File: src/components/Swap/SwapActionButton/AllowanceButton.tsx (hunk 4 above)
Comment (zzmp) [reply src/components/Swap/SwapActionButton/AllowanceButton.tsx:6]: Be sure to change the imports of `hooks/useTokenColorExtraction` to rename the import to `useTokenExtractionColor`.

**Review by zzmp — CHANGES_REQUESTED** [review]

File: src/components/Swap/SwapActionButton/AllowanceButton.tsx (hunk 4 above)
Comment (just-toby) [reply src/components/Swap/SwapActionButton/AllowanceButton.tsx:6]: wow vs code betrayed me with this one

File: src/components/Swap/SwapActionButton/AllowanceButton.tsx (hunk 5 above)
Comment (just-toby) [reply src/components/Swap/SwapActionButton/AllowanceButton.tsx:71]: yes, we have a couple callsites that don't use `useButtonColor`

**Review by zzmp — APPROVED** [review]

**Pull request merged**